### PR TITLE
feat(deequ_basic_operators):hasSize & hasUniqueness

### PR DESCRIPTION
**Pull request by Irenez753**

### CODE REVIEW

File: src/test/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/HasUniquenessTest.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  test("has_uniqueness on a non-unique field with level error should raise exception") {
+    val employeeData = Seq(
+      ("Maria", 1, "Smith", 111, 1111),
+      ("Josh", 1, "Smith", 222, 2222)
+    )
+    val level = "error"
+
+    val thrown = intercept[Exception] {
+      valideHasUniquenessOverDf(employeeData, Seq("id", "name"), Check.IsOne, level)
```
Comment (liebstein) [inline src/test/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/HasUniquenessTest.scala:42]: Can you define the function? it should be more clear this function is (double => boolean)

File: src/main/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/operators/HasSize.scala [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.yotpo.metorikku.metric.stepActions.dataQuality.operators
+
+import com.amazon.deequ.checks.Check
+import com.yotpo.metorikku.metric.stepActions.dataQuality.Operator
+
+class HasSize(level: Option[String], assertion: Long => Boolean) extends Operator(level = level) {
```
Comment (liebstein) [inline src/main/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/operators/HasSize.scala:6]: `assertion: Option[Long => Boolean]`

File: src/main/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/operators/Evaluator.scala [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.yotpo.metorikku.metric.stepActions.dataQuality.operators
+
+case class Evaluator() {
+
+    def sizeAssertion(operator: String, evaluatee: Long): Long => Boolean = operator match {
+        case "==" => {_ == evaluatee}
```
Comment (Doe-Ed) [inline src/main/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/operators/Evaluator.scala:6]: Spark also supports === (and it is different from ==)
https://stackoverflow.com/questions/39490236/difference-between-and-in-scala-spark

File: src/test/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/HasUniquenessTest.scala [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  private def valideHasUniquenessOverDf(employeeData: Seq[(String, Int, String, Int, Int)], columns: Seq[String],
+                                        fraction: Double, operator: String, level: String) = {
+    val sqlContext = sparkSession.sqlContext
+    val hasUniquenessCheck = new HasUniqueness(level = Some(level),columns = columns, fraction, operator)
+    val dqCheckDefinitionList = DataQualityCheckList(List[DataQualityCheck](DataQualityCheck(None, None, hasUniqueness = Some(hasUniquenessCheck))), None)
+    import sqlContext.implicits._
+
+    val dfName = "employee_data"
+    val df = employeeData.toDF(dfName, "id", "name", "fake", "fake2")
```
Comment (Doe-Ed) [inline src/test/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/HasUniquenessTest.scala:28]: first column is the employee_name isn't it?
Not that it matters but it confused me...

File: src/main/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/operators/Evaluator.scala [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package com.yotpo.metorikku.metric.stepActions.dataQuality.operators
+
+case class Evaluator() {
+
+    def sizeAssertion(operator: String, evaluatee: Long): Long => Boolean = operator match {
```
Comment (liebstein) [inline src/main/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/operators/Evaluator.scala:5]: Try to create assertion with T==> Boolean

File: src/main/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/operators/Evaluator.scala [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package com.yotpo.metorikku.metric.stepActions.dataQuality.operators
+
+case class Evaluator() {
```
Comment (liebstein) [inline src/main/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/operators/Evaluator.scala:3]: If needed then use 2 case classes

File: src/main/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/operators/HasSize.scala [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package com.yotpo.metorikku.metric.stepActions.dataQuality.operators
+
+import com.amazon.deequ.checks.Check
+import com.yotpo.metorikku.metric.stepActions.dataQuality.Operator
+
+class HasSize(level: Option[String], size: Long, operator: String) extends Operator(level = level) {
+
+  override def getCheck(level: String): Check = {
+    new Check(getLevel(level), "Size check for data frame size: %s".format(size)).hasSize(Evaluator().sizeAssertion(operator, size))
```
Comment (liebstein) [inline src/main/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/operators/HasSize.scala:9]: `%s".format(size)).hasSize(dqAssertion(operator, size).getAssertion())`

File: examples/dq/README.md [hunk 8]
```diff
@@ -35,6 +35,14 @@ Required parameters: `column`
...
 Required parameters: `column`
-
+##### `hasSize`
+Creates a constraint that calculates the data frame size and asserts the size according to the given operator defined.
+Required parameters: `size`
+Required parameters: `operator`
+##### `hasUniqueness`
+Creates a constraint that asserts on uniqueness in a single or combined set of key columns, according to the given fraction and operator.
+Required parameters: `columns`
+Required parameters: `fraction`
```
Comment (liebstein) [inline examples/dq/README.md:45]: Optional

File: src/main/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/operators/HasUniqueness.scala [hunk 9]
```diff
@@ -4,10 +4,10 @@ import com.amazon.deequ.checks.Check
 import com.yotpo.metorikku.metric.stepActions.dataQuality.Operator
 
 class HasUniqueness(level: Option[String], columns: Seq[String],
-                    fraction: Double, operator: String) extends Operator(level = level) {
+                    fraction: String, operator: String) extends Operator(level = level) {
```
Comment (liebstein) [inline src/main/scala/com/yotpo/metorikku/metric/stepActions/dataQuality/operators/HasUniqueness.scala:7]: change fraction and operator to option

File: examples/dq/README.md [hunk 10]
```diff
@@ -35,6 +35,14 @@ Required parameters: `column`
 ##### `isUnique`
 Creates a constraint that asserts on a column uniqueness.
 Required parameters: `column`
-
+##### `hasSize`
+Creates a constraint that calculates the data frame size and asserts the size according to the given operator defined.
+Required parameters: `size`
+Required parameters: `operator`
```
Comment (liebstein) [inline examples/dq/README.md:41]: write all optional values: `<,>,==,<= ....`

**Pull request merged**